src/proto/mod.rs

```diff
@@ -193,6 +260,47 @@ impl<S: Read + Write> Drop for Client<S> {
     }
 }
 
+#[cfg(feature = "ent")]
+#[cfg_attr(docsrs, doc(cfg(feature = "ent")))]
+impl<S: Read + Write> Client<S> {
+    /// Send information on a job's execution progress to Faktory.
+    pub fn set_progress(&mut self, upd: ProgressUpdate) -> Result<(), Error> {
+        let cmd = Track::Set(upd);
+        self.issue(&cmd)?.await_ok()
+    }
+
+    /// Fetch information on a job's execution progress from Faktory.
+    pub fn get_progress(&mut self, jid: String) -> Result<Option<Progress>, Error> {
+        let cmd = Track::Get(jid);
+        self.issue(&cmd)?.read_json()
+    }
+
+    /// Fetch information on a batch of jobs execution progress.
+    pub fn get_batch_status(&mut self, bid: String) -> Result<Option<BatchStatus>, Error> {
+        let cmd = GetBatchStatus::from(bid);
+        self.issue(&cmd)?.read_json()
+    }
+}
+
+impl Client<TcpStream> {
```

Comment:
I recommend moving this as the first `impl` block on `Client` so that it is more clearly visible in the generated docs.

Reply:
done